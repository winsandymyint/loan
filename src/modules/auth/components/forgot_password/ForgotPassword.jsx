import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Field } from 'redux-form'
import { Button, Icon, ReduxFormInput } from 'components'
import { email, required } from 'utils/formValidation'
import { success } from 'utils/message'
import { translate } from 'utils/translation'
import logoFull from 'assets/images/logo-full.png'

class ForgotPassword extends Component {
    componentWillReceiveProps = nextProps => {
        const { errorObject, isForgotPasswordSent, isLoadingForgotPassword } = nextProps

        if (this.props.isLoadingForgotPassword !== isLoadingForgotPassword && !isLoadingForgotPassword) {
            if (!errorObject && isForgotPasswordSent) {
                success(translate('AUTH_RESET_PASSWORD_SUCCESS'))
            }
        }
    }

    handleFormSubmit = ({ email }) => {
        this.props.forgotPassword(email)
    }

    handleGoBack = () => {
        this.props.history.push('/login')
    }

    render() {
        const { handleSubmit, isLoadingForgotPassword } = this.props

        return (
            <section className='forgot-password'>
                <form className='forgot-password-form' onSubmit={handleSubmit(this.handleFormSubmit)}>
                    <div className='img-container'>
                        <img src={logoFull} />
                    </div>
                    <Field
                        component={ReduxFormInput}
                        name='email'
                        props={{
                            disabled: isLoadingForgotPassword,
                            placeholder: translate('INPUT_EMAIL_PLACEHOLDER'),
                            prefix: <Icon type='mail' />,
                            type: 'email'
                        }}
                        validate={[required, email]}
                    />
                    <div className='btn-container'>
                        <Button
                            htmlType='submit'
                            icon='retweet'
                            loading={isLoadingForgotPassword}
                            name='login-btn'
                            type='primary'
                            value={translate('AUTH_RESET_PASSWORD_TEXT')}
                        />
                    </div>
                    <div className='btn-container'>
                        <Button
                            ghost
                            icon='left'
                            name='go-back-btn'
                            onClick={this.handleGoBack}
                            type='primary'
                            value={translate('GENERAL_RETURN_TEXT')}
                        />
                    </div>
                </form>
            </section>
        )
    }
}

ForgotPassword.propTypes = {
    errorObject: PropTypes.object,
    forgotPassword: PropTypes.func,
    handleSubmit: PropTypes.func,
    history: PropTypes.object,
    isForgotPasswordSent: PropTypes.bool,
    isLoadingForgotPassword: PropTypes.bool
}
export default ForgotPassword
