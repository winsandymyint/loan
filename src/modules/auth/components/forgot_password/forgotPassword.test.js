/* global test expect shallow */

import React from 'react'
import ForgotPassword from './ForgotPassword'

test('component matches snapshot', () => {
    const wrapper = shallow(<ForgotPassword handleSubmit={() => undefined} />)
    expect(wrapper).toMatchSnapshot()
})

test('render props not in root node', () => {
    const wrapper = shallow(<ForgotPassword handleSubmit={() => undefined} isLoading />)
    expect(wrapper.instance().props).toHaveProperty('isLoading', true)
})
