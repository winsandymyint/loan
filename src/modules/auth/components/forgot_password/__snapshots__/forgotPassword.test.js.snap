// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component matches snapshot 1`] = `
<section
  className="forgot-password"
>
  <form
    className="forgot-password-form"
  >
    <div
      className="img-container"
    >
      <img
        src="test-file-stub"
      />
    </div>
    <Field
      component={[Function]}
      name="email"
      props={
        Object {
          "disabled": undefined,
          "placeholder": "Enter your email",
          "prefix": <Icon
            spin={false}
            theme="outlined"
            type="mail"
          />,
          "type": "email",
        }
      }
      validate={
        Array [
          [Function],
          [Function],
        ]
      }
    />
    <div
      className="btn-container"
    >
      <Button
        disabled={false}
        htmlType="submit"
        icon="retweet"
        loading={false}
        name="login-btn"
        size="default"
        type="primary"
        value="Reset"
      />
    </div>
    <div
      className="btn-container"
    >
      <Button
        disabled={false}
        ghost={true}
        htmlType="button"
        icon="left"
        loading={false}
        name="go-back-btn"
        onClick={[Function]}
        size="default"
        type="primary"
        value="Go back"
      />
    </div>
  </form>
</section>
`;
